### PR TITLE
PI-2124: Added endpoint to find AP documents

### PR DESCRIPTION
**Pull request by pmcphee77**

### CODE REVIEW

File: projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/entity/Document.kt [hunk 1]
```diff
@@ -25,13 +19,46 @@ class Document(
...
+    val tableName: String,
+
+    @Column(name = "created_datetime")
+    val createdAt: ZonedDateTime,
+
+    @Column
+    val createdByUserId: Long,
+
+    @Column
+    val lastUpdatedUserId: Long,
```
[the remark this answers is not in the document]
Comment (pmcphee77) [reply projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/entity/Document.kt:44]: The last updated user is always used when its present an overrides the author. Thats the way it is in prisoner profile.

[the remark this answers is not in the document]
Comment (pmcphee77) [reply projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/entity/Document.kt:44]: added the integration tests as well

**Comment by marcus-bcl** [comment]
PR looks good, but do we definitely want all documents linked to AP Referrals?  Andre said "we just need the most recent document with the 'Item' type `AP Residence Plan Prepared`" (https://mojdt.slack.com/archives/C02HQ4M2YQN/p1713523133345399?thread_ts=1713185658.960839&cid=C02HQ4M2YQN).

There are a few template names relating to AP Residence Plans, I wonder if that's what they meant?  From what I can tell those residence plans aren't attached to the AP Referral in Delius.

**Comment by pmcphee77** [comment]
> PR looks good, but do we definitely want all documents linked to AP Referrals? Andre said "we just need the most recent document with the 'Item' type `AP Residence Plan Prepared`" (https://mojdt.slack.com/archives/C02HQ4M2YQN/p1713523133345399?thread_ts=1713185658.960839&cid=C02HQ4M2YQN).
> 
> There are a few template names relating to AP Residence Plans, I wonder if that's what they meant? From what I can tell those residence plans aren't attached to the AP Referral in Delius, they're typically added to a contact or an event.

Changed to use template name

**Pull request closed without merging**